src/index.ts

```diff
@@ -86,208 +144,262 @@ const allocationDatacap = async () => {
           {
             owner,
             repo,
             issue_number: issue.number,
-          }
-        );
-
-
-        //parse weeklhy dc in issue
-        // the amount to take into account is expected weekly usage rate or 5% of total dc requested (the lower)
-        // in this case I compare the entire weekly amount and 10% of total datacap requested
-        const totaldDcRequestedByClient = parseIssue(issue.body).datacapRequested
-        const weeklyAllocationRequestedByClient = parseIssue(issue.body).dataCapWeeklyAllocation
-        const weeklyDcAllocationBytes = anyToBytes(weeklyAllocationRequestedByClient.toString());
-        const totaldDcRequestedByClientBytes = anyToBytes(totaldDcRequestedByClient.toString());
-        // const tenPercentAllocationBytes = anyToBytes(totaldDcRequestedByClient.toString()) * 0.1;
-        // const allocation = weeklyDcAllocationBytes <= tenPercentAllocationBytes ? weeklyDcAllocationBytes : tenPercentAllocationBytes;
-        // const allocationRule = weeklyDcAllocationBytes <= tenPercentAllocationBytes ? "weekly" : "percentageDatacap";
-
-        //Parse dc requested msig notary address and  client address
-        const requestList = await findDatacapRequested(issueComments);
-        const lastRequest = requestList[requestList.length - 1];
-        const requestNumber = requestList.length;
-
-        if (lastRequest === undefined) {
-          logGeneral(`Issue number ${issue.number} skipped --> DataCap allocation requested comment is not present`);
-          continue;
-        }
-        if (!lastRequest.allocationDatacap && !lastRequest.clientAddress) {
-          logGeneral(`Issue number ${issue.number} skipped --> DataCap allocation requested comment is not present`);
-          continue;
-        }
-        if (!lastRequest.clientAddress) {
-          logGeneral(`Issue number ${issue.number} skipped --> clientAddress not found after parsing the comments`);
-          continue;
-        }
-        if (!lastRequest.allocationDatacap) {
-          logGeneral(`Issue number ${issue.number} skipped --> datacapAllocated not found after parsing the comments`);
-          continue;
-        }
-
-        //Check datacap remaining for this address
-
-        const client = clientsByVerifierRes.data.data.find((item: any) => item.address == lastRequest.clientAddress);
-        if (!client) {
-          logGeneral(`Issue number ${issue.number} skipped --> dc not allocated yet`);
-          continue;
-        }
-
-        const totalDcGrantedForClientSoFar = client.allowanceArray.reduce((s: number, item: any) => s + parseInt(item.allowance), 0)
-        // console.log("reduce", totalDcGrantedForClientSoFar)
+          })
+        resolve({ issueNumber: issue.number, comments })
+      })
+      ))
+
+    const requestListForEachIssue = await Promise.all(
+      commentsForEachIssue.map((issue: any) => {
+        return new Promise<any>(async (resolve, reject) => {
```

Comment:
Same here, not sure why you need to create a promise there

Reply:
fixed with last commit